core/src/kura.rs

```diff
@@ -319,17 +319,17 @@ impl Kura {
     }
 
     /// Put a block in kura's in memory block store.
-    pub fn store_block(&self, block: impl Into<Arc<VersionedCommittedBlock>>) {
-        let block = block.into();
-        self.block_data.lock().push((block.hash(), Some(block)));
+    pub fn store_block(&self, block: VersionedCommittedBlock) {
```

Comment:
It was `impl Into<Arc<VersionedCommittedBlock>>` on purpose to avoid redundant cloning.

Reply:
I avoided redundant clone with other means